### PR TITLE
Query API for event searches

### LINKED ISSUE #236: Optimize event searching
While we finally have functioning (😱!) search for events, it still has room for improvements, for example as of right now, it does not filter fetched events by event-type, so if you already have limited the event-types and want to expand your time-span, then it fetches every single event-type, this could be implemented in a simple way so that it only fetches more events when expanding the event-type-filter, or a more complex solution that saves the date-ranges for each event-type you have, and then fetches for the duration you do not currently have the types for. 

An optimisation for the search-input would be to make it only filter the previously filtered list if typing in more, or another earlier saved list if expanding

There are probably many more improvements to add, but this would probably help a bit

The relevant code is located in `src/events/hooks/useEventsRepoState.ts`

### LINKED ISSUE #363: /events might display empty although there are events
**Describe the bug**  

On direct access of events, the page is blank. However, there is still events in the Redux store.
I suspect this is due to the change of events in #348, where Debounced event list was tried implemented with Redux, but mostly left as is. On direct access, `filtered` is `true`, using the debounced list.
 
**To Reproduce**
1. Go to 'https://beta.online.ntnu.no/events' directly without accessing the frontpage
2. See the lack of events
3. Check Redux-store and see the events available for display

**Expected behavior**

A list of events should be rendered.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] Windows 10
 - Browser [e.g. chrome, safari] Firefox 
 - Version [e.g. 22] 75

**Additional context**

A quick fix is to remove the debounced list and purely use store, but that might disable the search.

### PR DESCRIPTION
**Pull request by oleast**
Uses the events-api for searching events instead of doing it front-end. Gives indexed search based on Watson, and is overall more pleasant and responsive to use.

Resolves #236 
Resolves #363

### CODE REVIEW

File: src/events/components/SearchModule/index.tsx [hunk 1]
```diff
@@ -1,82 +1,51 @@
...
+          ))}
+        </select>
       </div>
-    </>
+    </div>
   );
 };
 
+const selectCurrentPage = () => (state: State) => {
+  return state.events.search.page;
```
Comment (FluidSense) [inline src/events/components/SearchModule/index.tsx:48]: What is a page?
Is it 1 one page of events from the REST API?

File: src/events/components/ImageView/index.tsx [hunk 2]
```diff
@@ -29,51 +21,37 @@ const RIGHT = [
...
   );
 };
 
+const selectFrontPageEventIdsOfTypes = (eventTypes: EventTypeEnum[]) => (state: State) => {
+  const now = DateTime.local();
+  return eventSelectors
+    .selectAll(state)
+    .filter((event) => eventTypes.some((eventType) => event.event_type === eventType))
+    .filter((event) => DateTime.fromISO(event.start_date) >= now)
+    .sort((eventA, eventB) => eventA.start_date.localeCompare(eventB.start_date))
```
[the remark this answers is not in the document]
Comment (oleast) [reply src/events/components/ImageView/index.tsx:52]: Yes they are, but different views need different orderings.
The image view has events sorted chronologically, while the search and list views have them sorted with the latest first.

File: src/events/components/SearchModule/NextPageObserver.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+export const NextPageObserver: FC = () => {
+  const dispatch = useDispatch();
+  const isPending = useSelector(selectIsSearchPending());
+  const totalCount = useSelector(selectTotalCount());
+  const currentCount = useSelector(selectCurrentCount());
+  const [observer, ref] = useIntersection();
+  const [renderCount, setRenderCount] = useState(0); // TODO: Implement inside useIntersection somehow.
+
+  const isSeachCompleted = totalCount === currentCount;
```
[the remark this answers is not in the document]
Comment (oleast) [reply src/events/components/SearchModule/NextPageObserver.tsx:18]: I'll fix :smile:

File: src/events/components/SearchModule/index.tsx (hunk 1 above)
Comment (oleast) [reply src/events/components/SearchModule/index.tsx:48]: It is the pages from the API, yes. We're still using pagination in a way, but we're abstracting it as much as we can away from the user.

File: src/events/components/SearchModule/NextPageObserver.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (oleast) [reply src/events/components/SearchModule/NextPageObserver.tsx]: Done

**Pull request merged**